Lib/asyncio/proactor_events.py

```diff
@@ -627,7 +628,9 @@ def __init__(self, proactor):
         proactor.set_loop(self)
         self._make_self_pipe()
         self_no = self._csock.fileno()
-        signal.set_wakeup_fd(self_no)
+        if threading.current_thread() is threading.main_thread():
+            # wakeup fd can be installed from main thread only
```

Comment:
Minor clarification/nit: 

```suggestion
            # wakeup fd can only be installed to a file descriptor from the main thread
```